lib/portlayer/exec/container.go

```diff
@@ -209,6 +209,12 @@ func (c *Container) start(ctx context.Context) error {
 		return err
 	}
 
+	_, err = c.waitForPowerState(ctx, 10*time.Second, types.VirtualMachinePowerStatePoweredOn)
```

Comment:
I'm not happy with an arbitrary timeout - vSphere will start queuing operations when there's more than 60 so in a busy env this could take a while. In a nested environment it's worse.

Additionally `detail, err = c.vm.WaitForKeyInExtraConfig(ctx, key)` should have blocked until the process was launched - why is it insufficient?